File: bin/bob/examples/lib/android-java-lib/src/com/example/popup/Popup.java
package com.example.popup;

import android.app.AlertDialog;
import android.content.Context;

public class Popup {
    private final Context context;
    private final String text;

    public Popup(Context context, String text) {
        this.context = context;
        this.text = text;
    }

    public void show() {
        new AlertDialog.Builder(context)
            .setTitle(context.getString(R.string.popup_title))
            .setMessage(text)
            .setPositiveButton("OK", (dialog, id) -> dialog.dismiss())
            .create()
            .show();
    }
}

File: bin/bob/examples/lib/android-kotlin-lib/src/com/example/popup/Popup.kt
package com.example.popup

import android.app.AlertDialog
import android.content.Context

class Popup(private val context: Context, private val text: String) {
    fun show() {
        AlertDialog.Builder(context)
            .setTitle(context.getString(R.string.popup_title))
            .setMessage(text)
            .setPositiveButton("OK") { dialog, _ -> dialog.dismiss() }
            .create()
            .show()
    }
}
